chromadb/db/mixins/sysdb.py

```diff
@@ -0,0 +1,441 @@
+                )
+            )
+            sql, params = get_sql(insert_segment, self.parameter_format())
+            try:
+                cur.execute(sql, params)
+            except self.unique_constraint_error() as e:
+                raise UniqueConstraintError(
+                    f"Segment {segment['id']} already exists"
+                ) from e
+            metadata_t = Table("segment_metadata")
+            if segment["metadata"]:
+                self._insert_metadata(
+                    cur,
+                    metadata_t,
+                    metadata_t.segment_id,
+                    segment["id"],
+                    segment["metadata"],
+                )
+
+    @override
+    def create_collection(self, collection: Collection) -> None:
+        """Create a new collection"""
+        with self.tx() as cur:
+            collections = Table("collections")
+            insert_collection = (
+                self.querybuilder()
+                .into(collections)
+                .columns(collections.id, collections.topic, collections.name)
+                .insert(
+                    ParameterValue(self.uuid_to_db(collection["id"])),
+                    ParameterValue(collection["topic"]),
+                    ParameterValue(collection["name"]),
+                )
+            )
+            sql, params = get_sql(insert_collection, self.parameter_format())
+            try:
+                cur.execute(sql, params)
+            except self.unique_constraint_error() as e:
+                raise UniqueConstraintError(
+                    f"Collection {collection['id']} already exists"
+                ) from e
+            metadata_t = Table("collection_metadata")
+            if collection["metadata"]:
+                self._insert_metadata(
+                    cur,
+                    metadata_t,
+                    metadata_t.collection_id,
+                    collection["id"],
+                    collection["metadata"],
+                )
+
+    @override
+    def get_segments(
+        self,
+        id: Optional[UUID] = None,
+        type: Optional[str] = None,
+        scope: Optional[SegmentScope] = None,
+        topic: Optional[str] = None,
+        collection: Optional[UUID] = None,
+    ) -> Sequence[Segment]:
+        segments_t = Table("segments")
+        metadata_t = Table("segment_metadata")
+        q = (
+            self.querybuilder()
+            .from_(segments_t)
+            .select(
+                segments_t.id,
+                segments_t.type,
+                segments_t.scope,
+                segments_t.topic,
+                segments_t.collection,
+                metadata_t.key,
+                metadata_t.str_value,
+                metadata_t.int_value,
+                metadata_t.float_value,
+            )
+            .left_join(metadata_t)
+            .on(segments_t.id == metadata_t.segment_id)
+            .orderby(segments_t.id)
+        )
+        if id:
```

Comment:
Can we explicitly check for None as opposed to relying on the falsiness of the value. For these types I think the only weird case is the empty string but still worth the cleanliness IMO